helm.rst

```diff
@@ -0,0 +1,161 @@
+   * - producers
+     - This section holds the configuration of the individual producers [#]_
+   * - producers.name
+     - This key gives a name to the producer deployment and can be repeated
+   * - producers.name.cscs [#]_
+     - The list of CSCs that the named producer will monitor
+   * - producers.name.image
+     - This section provides optional override of the default image section
+   * - producers.name.image.repository
+     - The Docker registry container image name to use for the named producer
+   * - producers.name.image.tag
+     - The container image tag to use for the named producer
+   * - producers.name.image.pullPolicy
+     - The policy to apply when pulling an image for named producer deployment
+   * - producers.name.env
+     - This section provides optional override of the defaults env section
+   * - producers.name.env.lsstDdsDomain
+     - The LSST_DDS_DOMAIN name applied the named producer container
+   * - producers.name.env.partitions
+     - The number of partitions that the named producer is supporting
+   * - producers.name.env.replication
+     - The number of replications available to the named producer
+   * - producers.name.env.logLevel
+     - This value determines the logging level for the named producer     
+
+.. [#] A given producer is given a name key that is used to identify that producer (e.g. auxtel).
+.. [#] The characters >- are used after the key so that the CSCs can be specified in a list
+
+.. NOTE:: The brokerIp, brokerPort and registryAddr of the env section are not overrideable in the producers.name.env section. Control of those items is on a site basis. All producers at a given site will always use the same information.
+
+CSC Chart
+---------
+
+Instead of having charts for every CSC, we employ an approach of having one chart that describes all the different CSC variants. There are four main variants that the chart supports:
+
+simple
+  A CSC that requires no special interventions and uses only environment variables for configuration
+
+entrypoint
+  A CSC that uses an override script for the container entrypoint.
+
+imagePullSecrets
+  A CSC that requires the use of the Nexus3 repository and need access credential for pulling the associated image
+
+volumeMount
+  A CSC that requires access to a physical disk store in order to transfer information into the running container
+
+The chart consists of the Job Kubernetes Workflows API, ConfigMap and PersistentVolumeClaim Kubernetes Config and Storage APIs and VaultSecret `Vault <https://www.vaultproject.io/>`_ API. The Job API is used to provide correct behavior when a CSC is sent of OFFLINE mode, the pod should not restart. The drawback to this is if a CSC dies for an unknown reason, not one caught by FAULT state transition, the pod will not restart and requires startup intervention. The other three APIs are used to support the non-simple CSC variants. They will be mentioned in the configuration description which we will turn to next.
```

Comment:
> The Job API is used to provide correct behavior when a CSC is sent of OFFLINE mode, the pod should not restart. The drawback to this is if a CSC dies for an unknown reason, not one caught by FAULT state transition, the pod will not restart and requires startup intervention.

I wonder if this is something we could improve in the future?

Reply:
I found the way to get this to work. Now if the CSC dies unexpectedly, a new pod will start in it's place, but the old one will remain around for debugging. You will still have to bring the component out of STANDBY, but we now get automatic restart in case of unknown failures. I will fix the text.